go/obscuronode/enclave/enclave.go

```diff
@@ -70,6 +77,70 @@ type enclaveImpl struct {
 	blockProcessingMutex sync.Mutex
 }
 
+// NewEnclave creates a new enclave.
+// `genesisJSON` is the configuration for the corresponding L1's genesis block. This is used to validate the blocks
+// received from the L1 node if `validateBlocks` is set to true.
+func NewEnclave(
```

Comment:
If you'll forgive some minor refactoring, I moved the `NewEnclave` method to the top of the file, for consistency with other files. Mea maxima culpa.

Reply:
Looking at this method and trying to figure out what if anything has changed I have a new appreciation for that rule haha